infra/aws/terraform/cncf-k8s-infra-aws-capa-ami/oidc.tf

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+module "iam_github_oidc_provider" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-github-oidc-provider"
+
+  client_id_list = [
+    "sts.amazonaws.com",
+  ]
+
+  tags = var.tags
+}
+
+module "iam_github_oidc_role" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-github-oidc-role"
+
+  name = "gh-image-builder"
+  subjects = ["kubernetes-sigs/cluster-api-provider-aws:*"]
```

Comment:
Is there a particular reason why we are using github actions instead of prow? We can configure prow jobs to assume AWS roles very easily

Reply:
When we started this work in was advised to use TF.  Do we use IRSA or pod identity with any Prow jobs at present?

Reply:
Having a quick look i only see assume roles ina few jobs and these are "trusted" jobs.

Reply:
We do, with kops and capz.

Reply:
Excellent, i will hunt them out. Thanks @upodroid

Reply:
@upodroid - any chance you could point me at one of the jobs? I'm not really seeing how this is done from Prow. 

I can see this trusted job: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-k8s-infra/trusted/sig-k8s-infra-registry.yaml#L32

Reply:
@upodroid - any chance you could point me at one of the existing jobs that use assume roles?